kasa/smart/smartdevice.py

```diff
@@ -320,6 +320,17 @@ async def _handle_modular_update_error(
                 responses[meth] = SmartErrorCode.INTERNAL_QUERY_ERROR
         return responses
 
+    def _required_key_on_parent(self, mod: type[SmartModule]) -> bool:
```

Comment:
Is `REQUIRED_KEY_ON_PARENT` badly named? afaict only the contact sensor is using `REQUIRED_KEY_ON_PARENT` and it has the key in it's own sysinfo rather than the parent. Maybe this was something we used before we had child components in the parent query but it feels like this should be called `KEYS_IN_INFO`?

If we do rename we might as well change the contact sensor to use a list and can do away with the method,

Reply:
Yeah, this is indeed badly named. Looks like you also predicted the need for having this as a list in https://github.com/python-kasa/python-kasa/pull/877#discussion_r1591959655 :-)

How about `REQUIRED_KEY_IN_INFO`? Or `WANTED_KEY_IN_INFO`? Even though, neither describes that only a single one should match..

Reply:
Renamed to `SYSINFO_LOOKUP_KEYS` which sounded somehow better to me, but I can still change it. 

Also, added some tests for `SmartDevice._initialize_modules` for both this and for the component availability initialization.